nixos/modules/security/apparmor.nix

```diff
@@ -29,6 +39,577 @@ in
    config = mkIf cfg.enable {
      environment.systemPackages = [ pkgs.apparmor-utils ];
 
+     # Needed globals and abstractions that can be included in profiles
+     environment.etc."/apparmor.d/abstractions".source = pkgs.apparmor-profiles + "/etc/apparmor.d/abstractions";
+     environment.etc."/apparmor.d/tunables".source = pkgs.apparmor-profiles + "/etc/apparmor.d/tunables";
+
+     environment.etc."apparmor/parser.conf".text = ''
+      Optimize=compress-fast
+      ${cfg.extraParserConfig}
+    '';
+
+     # Pretty much everything in here, is there to
+     # prevent apparmor utilities from crashing.
+     environment.etc."apparmor/logprof.conf".text =
+      builtins.replaceStrings
+      [
+        "/usr"
+      ]
+      [
+        "/nix/store/*"
```

Comment:
I am not yet sure about the implications of this.
In NixOS not just root can install stuff to the /nix/store...

Reply:
These configuration files do not affect the behaviour of apparmor itself and only affect the apparmor-utils. They use the [globs] in parser.conf to add asterisks where needed when they generate a rule for you through aa-logprof.
The severity.db will be used by aa-logprof to grade the severity of binaries and display them to the user.